python/flask/security/unescaped-template-extension.yaml

```diff
@@ -0,0 +1,33 @@
+rules:
+- id: unescaped-template-extension
+  patterns:
+  - pattern-not: flask.render_template("=~/.+\.html$/", ...)
+  - pattern-not: flask.render_template("=~/.+\.xml$/", ...)
+  - pattern-not: flask.render_template("=~/.+\.htm$/", ...)
+  - pattern-not: flask.render_template("=~/.+\.xhtml$/", ...)
+  - pattern-not: flask.render_template($X + "=~/\.html$/", ...)
+  - pattern-not: flask.render_template($X + "=~/\.xml$/", ...)
+  - pattern-not: flask.render_template($X + "=~/\.htm$/", ...)
+  - pattern-not: flask.render_template($X + "=~/\.xhtml$/", ...)
```

Comment:
Is `=~/[x]?[xh]tm[l]?` worth it?

Reply:
I tried that, but the Ocaml regex engine seemed to not comprehend it: https://semgrep.live/8ej

Reply:
And with backslashes everywhere: https://semgrep.live/gze